cmd/traffic/cmd/manager/internal/mutator/watcher.go

```diff
@@ -389,66 +399,123 @@ func (c *configWatcher) Store(ctx context.Context, ac *agentconfig.Sidecar, upda
 		return nil
 	}
 
-	create := false
-	api := k8sapi.GetK8sInterface(ctx).CoreV1().ConfigMaps(ns)
-	cm, err := api.Get(ctx, agentconfig.ConfigMap, meta.GetOptions{})
-	if err != nil {
-		if !errors.IsNotFound(err) {
-			return fmt.Errorf("unable to get ConfigMap %s: %w", agentconfig.ConfigMap, err)
+	c.configUpdatersMapLock.Lock()
+	var cu *configUpdater
+
+	newGroup := false
+	if _, ok := c.configUpdaters[ns]; !ok {
+		grp, _ := errgroup.WithContext(ctx)
+		c.configUpdaters[ns] = &configUpdater{
+			grp,
+			map[string]string{},
+			map[string]struct{}{},
 		}
-		create = true
-	} else {
-		// Ensure that we're not about to overwrite a manually added config entry
-		if currentYml, ok := cm.Data[ac.AgentName]; ok {
-			var currAc agentconfig.Sidecar
-			if err = decode(currentYml, &currAc); err == nil && currAc.Manual {
-				dlog.Warnf(ctx, "avoided an attempt to overwrite manually added config entry for %s.%s", ac.AgentName, ns)
-				return nil
+		newGroup = true
+	}
+
+	cu = c.configUpdaters[ns]
+	cu.Config[ac.AgentName] = yml
+
+	if updateSnapshot {
+		cu.AddToSnapshot[ac.AgentName] = struct{}{}
+	}
+
+	c.configUpdatersMapLock.Unlock()
```

Comment:
Why isn't this unlocked on line 418? Does this lock also protect the actual `cu`?